### DISCUSSION TITLE
Are these type errors expected with createPages?

### DISCUSSION THREAD

**Discussion by sohcfst** [wakujs/waku#553] [Q&A] [answered]
this does not affect actual developement, other than not being able to ignore red squiggly lines but was wondering! trying out waku today and excited to use it for a landing page.

```
createPages.d.ts(29, 5): The expected type comes from property 'component' which is declared here on type '{ render: "static"; path: "/"; component: FunctionComponent<RouteProps & { children: ReactNode; }>; }'
```

<img width="863" alt="Screenshot 2024-02-29 at 3 18 29 PM" src="https://github.com/dai-shi/waku/assets/67526712/f8474e61-26b7-43b0-98ce-c103df153383">

**Comment by sohcfst** [comment]
removing the .js brings this up: 

<img width="1055" alt="Screenshot 2024-02-29 at 3 21 49 PM" src="https://github.com/dai-shi/waku/assets/67526712/98a0a624-de88-4f13-8ebb-75ce482166e3">

**Comment by dai-shi** [comment, marked as answer]
Does it occur with the unmodified template?

The createPages type is full of hacks, and it's not too surprising if there are some bugs. But, it's at least working for the template.

**Reply by sohcfst** [reply]
yeah, unmodified. its a fresh init, inside of turborepo.

that's ok! i just wanted to make sure. thank you for your hard work. im very excited

**Reply by sohcfst** [reply]
interesting. i init outside of turborepo and types are fine. 

here's the diff

tsconfig for turborepo:

```
File: base.json
01: {
02:   "$schema": "https://json.schemastore.org/tsconfig",
03:   "display": "Default",
04:   "compilerOptions": {
05:     "declaration": true,
06:     "declarationMap": true,
07:     "esModuleInterop": true,
08:     "incremental": false,
09:     "isolatedModules": true,
10:     "lib": ["es2022", "DOM", "DOM.Iterable"],
11:     "module": "NodeNext",
12:     "moduleDetection": "force",
13:     "moduleResolution": "NodeNext",
14:     "noUncheckedIndexedAccess": true,
15:     "resolveJsonModule": true,
16:     "skipLibCheck": true,
17:     "strict": true,
18:     "target": "ES2022"
19:   }
20: }
```

and waku's

```
File: tsconfig.json
01: {
02:   "compilerOptions": {
03:     "strict": true,
04:     "target": "esnext",
05:     "downlevelIteration": true,
06:     "esModuleInterop": true,
07:     "module": "nodenext",
08:     "skipLibCheck": true,
09:     "noUncheckedIndexedAccess": true,
10:     "exactOptionalPropertyTypes": true,
11:     "types": ["react/experimental"],
12:     "jsx": "react-jsx",
13:     "paths": {
14:       "~/*": ["./src/*"]
15:     }
16:   }
17: }
```

**Reply by sohcfst** [reply]
thank you for being my rubber duck!
